plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorDubbingCatalogItem.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @package plugins.reach
+ * @subpackage api.objects
+ */
+class KalturaVendorDubbingCatalogItem extends KalturaVendorCatalogItem
+{
+	
+	/**
+	 * @var int
+	 */
+	public $flavorParamsId;
+	
+	/**
+	 * @var int
+	 */
+	public $clearAudioFlavorParamsId;
+	
+	private static $map_between_objects = array
+	(
+		'flavorParamsId',
+		'clearAudioFlavorParamsId',
+	);
+	
+	protected function getServiceFeature()
+	{
+		return KalturaVendorServiceFeature::DUBBING;
+	}
+	
+	/* (non-PHPdoc)
+	 * @see KalturaCuePoint::getMapBetweenObjects()
+	 */
+	public function getMapBetweenObjects()
+	{
+		return array_merge(parent::getMapBetweenObjects(), self::$map_between_objects);
+	}
+	
+	/* (non-PHPdoc)
+ * @see KalturaObject::toInsertableObject()
+ */
+	public function toInsertableObject($object_to_fill = null, $props_to_skip = array())
+	{
+		if (is_null($object_to_fill))
+			$object_to_fill = new VendorDubbingCatalogItem();
+		
+		return parent::toInsertableObject($object_to_fill, $props_to_skip);
+	}
+	
+	public function validateForInsert($propertiesToSkip = array())
+	{
+		$this->validatePropertyNotNull(array("sourceLanguage"));
+		$this->validatePropertyNotNull(array("flavorParamsId"));
+		$this->validatePropertyNotNull(array("clearAudioFlavorParamsId"));
+		
+		$this->validateServiceType();
+		$this->validateFlavorParamsId($this->flavorParamsId);
+		$this->validateFlavorParamsId($this->clearAudioFlavorParamsId);
+		
+		return parent::validateForInsert($propertiesToSkip);
+	}
+	
+	public function validateForUpdate($sourceObject, $propertiesToSkip = array())
+	{
+		/* @var $sourceObject VendorDubbingCatalogItem */
+		if(isset($this->flavorParamsId) && $this->flavorParamsId != $sourceObject->getFlavorParamsId())
+		{
+			$this->validateFlavorParamsId($this->flavorParamsId);
+		}
+		
+		if(isset($this->clearAudioFlavorParamsId) && $this->clearAudioFlavorParamsId != $sourceObject->getClearAudioFlavorParamsId())
+		{
+			$this->validateFlavorParamsId($this->clearAudioFlavorParamsId);
+		}
+		
+		if(isset($this->serviceType) && $this->serviceType!= $sourceObject->getServiceType())
+		{
+			$this->validateServiceType();
+		}
+		
+		return parent::validateForUpdate($sourceObject, $propertiesToSkip);
+	}
+	
+	protected function validateFlavorParamsId($id)
+	{
+		$flavorParams = assetParamsPeer::retrieveByPK($id);
+		if (!$flavorParams)
+			throw new KalturaAPIException(KalturaErrors::FLAVOR_PARAMS_ID_NOT_FOUND, $id);
+	}
+	
+	protected function validateServiceType()
+	{
+		if($this->serviceType != KalturaVendorServiceType::HUMAN)
+			throw new KalturaAPIException(KalturaReachErrors::CATALOG_ITEM_ONLY_HUMAN_ALLOWED);
```

Comment:
are you sure only human is allowed for dubbing?
also {}

Reply:
not sure. i adapted the Audio Description catalog item example